packages/serverpod/lib/src/server/session.dart

```diff
@@ -12,10 +13,30 @@ import 'package:serverpod/src/server/serverpod.dart';
 import '../cache/caches.dart';
 import '../database/database.dart';
 
+/// A listener that will be called when the session is about to close.
+typedef WillCloseListener = FutureOr<void> Function(Session);
+
 /// When a call is made to the [Server] a [Session] object is created. It
 /// contains all data associated with the current connection and provides
 /// easy access to the database.
 abstract class Session {
+  /// List of listeners that will be called when the session is about to close.
```

Comment:
Nit. Technically, the `///` have no effect for private methods.

Reply:
True, I can remove this! 👍